### PR TITLE
feat: add prototracks-to-seeds algorithm

### PR DESCRIPTION
**Pull request by benjaminhuth**
This allows to transform a prototracks collection into a seeds collection. Useful for the Exa.TrkX workflow.

### CODE REVIEW

File: Examples/Algorithms/Utilities/src/PrototracksToSeeds.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  m_inputProtoTracks.initialize(m_cfg.inputProtoTracks);
+  m_inputSpacePoints.initialize(m_cfg.inputSpacePoints);
+}
+
+ProcessCode PrototracksToSeeds::execute(const AlgorithmContext& ctx) const {
+  auto prototracks = m_inputProtoTracks(ctx);
+
+  prototracks.erase(std::remove_if(prototracks.begin(), prototracks.end(),
+                                   [](const auto& t) { return t.size() < 3; }),
+                    prototracks.end());
```
Comment (andiwand) [inline Examples/Algorithms/Utilities/src/PrototracksToSeeds.cpp:34]: should we add a logging that will tell you how much stuff got removed?

File: Examples/Framework/include/ActsExamples/Utilities/EventDataTransforms.hpp [hunk 2]
```diff
@@ -15,6 +15,16 @@ namespace ActsExamples {
...
+    Index index, const SimSpacePointContainer &spacepoints);
+
+class ProtoTrackToSeed {
+  const SimSpacePointContainer &m_spacePoints;
+
+ public:
+  ProtoTrackToSeed(const SimSpacePointContainer &spacepoints)
+      : m_spacePoints(spacepoints) {}
+  SimSeed operator()(const ProtoTrack &track) const;
+};
```
Comment (andiwand) [inline Examples/Framework/include/ActsExamples/Utilities/EventDataTransforms.hpp:28]: I wonder if a function would be nicer. for the `std::transform` you could call it through a lambda

**Review by andiwand — APPROVED** [review]

File: Examples/Algorithms/Utilities/src/PrototracksToSeeds.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
```
Comment (AJPfleger) [inline Examples/Algorithms/Utilities/src/PrototracksToSeeds.cpp:3]:
```suggestion
// Copyright (C) 2023 CERN for the benefit of the Acts project
```

File: Examples/Algorithms/Utilities/include/ActsExamples/Utilities/PrototracksToSeeds.hpp [hunk 4]
```diff
@@ -0,0 +1,54 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
```
Comment (AJPfleger) [inline Examples/Algorithms/Utilities/include/ActsExamples/Utilities/PrototracksToSeeds.hpp:3]:
```suggestion
// Copyright (C) 2023 CERN for the benefit of the Acts project
```

**Review by andiwand — APPROVED** [review]

**Pull request merged**